scripts/pca_variant_filter.py

```diff
@@ -0,0 +1,442 @@
+        min_joint_af,
+        min_joint_callrate,
+        min_ccdg_exome_callrate,
+        min_ukbb_exome_callrate,
+    )
+
+    variant_filter_expr = True
+    if bi_allelic_only:
+        variant_filter_expr &= ht.joint_biallelic
+    if min_inbreeding_coeff_threshold is not None:
+        variant_filter_expr &= (
+            ht.ccdg_genomes_site_inbreeding_coeff > min_inbreeding_coeff_threshold
+        ) & (ht.gnomad_genomes_site_inbreeding_coeff > min_inbreeding_coeff_threshold)
+    if min_hardy_weinberg_threshold is not None:
+        variant_filter_expr &= (
+            ht.ccdg_genomes_hwe.p_value > min_hardy_weinberg_threshold
+        ) & (ht.gnomad_genomes_hwe.p_value > min_hardy_weinberg_threshold)
+
+    variant_filter_expr &= (
+        (ht.joint_AF > min_joint_af)
+        & (ht.joint_callrate > min_joint_callrate)
+        & (ht.ccdg_exomes_AN / (ccdg_exome_count * 2) > min_ccdg_exome_callrate)
+        & (ht.ukbb_AN / (ukbb_exome_count * 2) > min_ukbb_exome_callrate)
+    )
+
+    ht = ht.filter(variant_filter_expr)
+
+    ht = ht.annotate_globals(
+        autosomes_only=autosomes_only,
+        snv_only=snv_only,
+        bi_allelic_only=bi_allelic_only,
+        min_gnomad_v3_ac=min_gnomad_v3_ac,
+        high_qual_ccdg_exome_interval_only=high_qual_ccdg_exome_interval_only,
+        high_qual_ukbb_exome_interval_only=high_qual_ukbb_exome_interval_only,
+        filter_lcr=filter_lcr,
+        filter_segdup=filter_segdup,
+        min_af=min_joint_af,
+        min_callrate=min_joint_callrate,
+        min_ccdg_exome_callrate=min_ccdg_exome_callrate,
+        min_ukbb_exome_callrate=min_ukbb_exome_callrate,
+        min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
+        min_hardy_weinberg_threshold=min_hardy_weinberg_threshold,
+    )
+
+    ht = filter_low_conf_regions(
+        ht,
+        filter_lcr=filter_lcr,
+        filter_decoy=False,  # No decoy for GRCh38
+        filter_segdup=filter_segdup,
+    )
+
+    ht.checkpoint(
+        get_pca_variants_path(ld_pruned=False),
+        overwrite=overwrite,
+        _read_if_exists=(not overwrite),
+    )
+
+    if ld_pruning:
+        logger.info("Creating Table after LD pruning of %s...", ld_pruning_dataset)
+        ht = hl.read_table(get_pca_variants_path(ld_pruned=False))
+        if ld_pruning_dataset == "ccdg_genomes":
+            vds = hl.vds.read_vds(get_ccdg_vds_path("genomes"))
+            vds = hl.vds.split_multi(vds)
+            vds = hl.vds.filter_variants(vds, ht, keep=True)
+            mt = hl.vds.to_dense_mt(vds)
+        elif ld_pruning_dataset == "gnomad_genomes":
+            mt = get_gnomad_v3_mt(split=True)
+            mt = densify_sites(mt, ht, hl.read_table(gnomad_last_END_position.path))
+            mt = filter_ref_blocks(mt)
+        else:
+            ValueError(
+                "Only options for LD pruning are `ccdg_genomes` and `gnomad_genomes`"
+            )
+
+        ht = hl.ld_prune(mt.GT, r2=ld_r2)
```

Comment:
I'm not sure if we will still have a problem, but I know when I needed to do this ld-prune on one of the earlier UKBB releases I needed to use all workers. https://github.com/broadinstitute/gnomad_methods/blob/35066ffc01d63ac2d7b20e069ea6703013ae9da7/gnomad/sample_qc/pipeline.py#L153